hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/CallerContext.java

```diff
@@ -108,24 +117,86 @@ public String toString() {
   }
 
   /** The caller context builder. */
-  public static final class Builder {
-    private final String context;
+  public static final class Builder implements Configurable {
+    private String context;
     private byte[] signature;
+    private Configuration conf = new Configuration();
 
     public Builder(String context) {
       this.context = context;
     }
 
+    public Builder(String key, String value, String kvSeparator) {
+      this.context = new StringBuilder(key)
+          .append(kvSeparator)
+          .append(value)
+          .toString();
+    }
+
     public Builder setSignature(byte[] signature) {
       if (signature != null && signature.length > 0) {
         this.signature = Arrays.copyOf(signature, signature.length);
       }
       return this;
     }
 
+    /**
+     * Append new item to the context.
+     * @param item
+     * @return builder
+     */
+    public Builder append(String item) {
+      if (CallerContext.isContextValid(context)) {
+        this.context = new StringBuilder(context)
+            .append(conf.get(HADOOP_CALLER_CONTEXT_ITEM_SEPARATOR_KEY,
+                HADOOP_CALLER_CONTEXT_ITEM_SEPARATOR_DEFAULT))
+            .append(item)
+            .toString();
+      } else {
+        this.context = item;
+      }
+      return this;
+    }
+
+    /**
+     * Append new item which contains key and value to the context.
+     * @param key
+     * @param value
+     * @param kvSeparator
+     * @return builder
+     */
+    public Builder append(String key, String value, String kvSeparator) {
```

Comment:
Thank you @ferhui for providing the patch.
Are there any use cases for this method instead of calling `append(key + kvSeparator + value)` directly?

Reply:
Now there isn't. Want to use append in RBF, for example, add callerip to context, append("CallerIp", "x.x.x.x", ":"). If this method is useless， we can remove it.